File: benchmarks/pandas/bench_series_to_markdown.py
"""
Benchmark: Series.to_markdown() and Series.to_latex() on a 500-element numeric Series.

Mirrors tsb seriesToMarkdown and seriesToLaTeX.
Outputs JSON: {"function": "series_to_markdown", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import pandas as pd

SIZE = 500
WARMUP = 5
ITERATIONS = 50

s = pd.Series([(i * 1.7) % 100 for i in range(SIZE)], name="values")

for _ in range(WARMUP):
    s.to_markdown()
    s.to_latex()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.to_markdown()
    s.to_latex()
total_ms = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_to_markdown",
    "mean_ms": total_ms / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total_ms,
}))

File: benchmarks/tsb/bench_series_to_markdown.ts
/**
 * Benchmark: seriesToMarkdown and seriesToLaTeX on a 500-element numeric Series.
 *
 * The existing `to_markdown` benchmark covers DataFrames only.
 * This benchmark exercises the Series variants: seriesToMarkdown / seriesToLaTeX.
 * Mirrors pandas Series.to_markdown() and Series.to_latex().
 *
 * Outputs JSON: {"function": "series_to_markdown", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series, seriesToMarkdown, seriesToLaTeX } from "../../src/index.ts";

const SIZE = 500;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({
  data: Array.from({ length: SIZE }, (_, i) => (i * 1.7) % 100),
  name: "values",
});

for (let i = 0; i < WARMUP; i++) {
  seriesToMarkdown(s);
  seriesToLaTeX(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesToMarkdown(s);
  seriesToLaTeX(s);
}
const total_ms = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_to_markdown",
    mean_ms: total_ms / ITERATIONS,
    iterations: ITERATIONS,
    total_ms,
  }),
);
